CONTRIBUTING.md

```diff
@@ -0,0 +1,125 @@
+
+## Pre-reqs
+
+- OS
+    - MacOS (or) Windows10
+- Language Runtimes
+    - .NET Core 2.0
+    - Python 3.6.x
+- Editor
+    - VS Code (or) Visual Studio
+- Tools
+    - [Azurite V2](https://github.com/Azure/Azurite/tree/legacy-master) (for MacOS) (or) [Azure Storage Emulator](https://docs.microsoft.com/en-us/azure/storage/common/storage-use-emulator) (or) Storage account in Azure
+    - [Azure Functions Core Tools](https://github.com/Azure/azure-functions-core-tools) v2.7.x and above.
+
+## Change flow
+
+The general flow for making a change to the script host is:
+1. 🍴 Fork the repo (add the fork via `git remote add me <clone url here>`
+2. 🌳 Create a branch for your change (generally use dev) (`git checkout -b my-change`)
+3. 🛠 Make your change
+4. ✔️ Test your changes
+5. ⬆️ Push your changes to your fork (`git push me my-change`)
+6. 💌 Open a PR to the dev branch
+7. 📢 Address feedback and make sure tests pass (yes even if it's an "unrelated" test failure)
+8. 📦 [Rebase](https://git-scm.com/docs/git-rebase) your changes into a meaningful commits (`git rebase -i HEAD~N` where `N` is commits you want to squash)
+9. :shipit: Rebase and merge (This will be done for you if you don't have contributor access)
+10. ✂️ Delete your branch (optional)
+
+## End to End Development Setup: JavaScript library + Durable Extension (MacOS)
+
+### Setting up Azurite V2
+
+Note: [Azurite V3](https://github.com/Azure/Azurite) does not have support for Table Storage yet. So falling back to [Azurite V2](https://github.com/Azure/Azurite/tree/legacy-master) setup.
+
+Create a folder say AzureWebJobsStorage
+
+`npm install azurite@2.7.1 -g`
+
+`azurite -l ./AzureWebJobsStorage`
+
+```
+ _______                   _             
+(_______)                 (_)  _         
+ _______ _____ _   _  ____ _ _| |_ _____ 
+|  ___  (___  ) | | |/ ___) (_   _) ___ |
+| |   | |/ __/| |_| | |   | | | |_| ____|
+|_|   |_(_____)____/|_|   |_| \__)_____)
+                                         
+Azurite, Version 2.7.1
+A lightweight server clone of Azure Storage
+
+Azure Table Storage Emulator listening on port 10002
+Azure Queue Storage Emulator listening on port 10001
+Azure Blob Storage Emulator listening on port 10000
+```
+
+### Visual Studio Code Extensions
+
+The following extensions should be installed if using Visual Studio Code for debugging:
+
+- C# for Visual Studio Code (powered by OmniSharp)
+- Azure Functions Extensions for Visual Studio Code v0.19.1 and above.
+
+### Setting up durable-js debugging
+
+- Create a Durable Functions Orchestrator for FunctionChaining pattern using [starter templates](https://docs.microsoft.com/en-us/azure/azure-functions/durable/quickstart-js-vscode)
+  Note: In this starter template, ignore the line that says: "On a Mac or Linux computer, you must set the AzureWebJobsStorage property to the connection string of an existing Azure storage account". We will be setting up the AzureWebJobsStorage property to `UseDevelopmentStorage=true`
```

Comment:
If you found that Azurite is sufficient for local dev, it may be a good idea for us to update this line in our template to give that option.

Reply:
Azurite V2 (which is not the version officially recommended by Azurite) works. V3 does not support some storage mechanisms. Does it make sense to add to starter templates now or wait until Azurite stabilizes?

Reply:
I see. It looks like there are no plans to support Tables in V3, due to Cosmos DB having a separate emulator for Azure Tables (which unfortunately, is NOT cross platform). You can see the [discussion here](https://github.com/Azure/Azurite/issues/253). With that in mind, maybe Azurite V2 is the best local dev option?

Reply:
@ConnorMcMahon : Yes given those limitations v2 is the best (as of now)